### PR TITLE
Fix transition in LA when handling canceled child wf

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
Fix transition in LA when handling canceled child wf/

closes https://github.com/temporalio/sdk-java/issues/2155

### CODE REVIEW

File: temporal-sdk/src/main/java/io/temporal/internal/statemachines/LocalActivityStateMachine.java [hunk 1]
```diff
@@ -121,6 +121,13 @@ enum State {
                   State.REQUEST_PREPARED,
                   LocalActivityStateMachine::sendRequest)
               .add(State.REQUEST_PREPARED, ExplicitEvent.MARK_AS_SENT, State.REQUEST_SENT)
+              // This is to cover am edge case where the event loop is
```
Comment (Sushisource) [inline temporal-sdk/src/main/java/io/temporal/internal/statemachines/LocalActivityStateMachine.java:124]:
```suggestion
              // This is to cover an edge case where the event loop is
```

File: temporal-sdk/src/test/java/io/temporal/internal/statemachines/LocalActivityStateMachineTest.java [hunk 2]
```diff
@@ -343,4 +348,99 @@ protected void buildWorkflow(AsyncWorkflowBuilder<Void> builder) {
...
+                    .setActivityId("id1")
+                    .setActivityType(ActivityType.newBuilder().setName("activity1")),
+                null,
+                System.currentTimeMillis(),
+                null,
+                false,
+                null);
+        // TODO: This is a workaround for the lack of support for child workflow in the test
+        // framework.
+        // The test framework has not support for state machines with multiple callbacks.
```
Comment (Sushisource) [inline temporal-sdk/src/test/java/io/temporal/internal/statemachines/LocalActivityStateMachineTest.java:373]:
```suggestion
        // The test framework has no support for state machines with multiple callbacks.
```

**Review by Sushisource — APPROVED** [review]
